### PR TITLE
Main to listing improvements

### PR DESCRIPTION
**Pull request by EPoikans**
Listing improvements and compatibility with CSRF token

### CODE REVIEW

File: source/controllers/ListingController.ts [hunk 1]
```diff
@@ -28,36 +30,46 @@ listing.read = async (req, res) => {
...
+				res.redirect('/listing/' + String(req.body.subcatid));
+			}
 		} catch (error) {
 			res.send(error);
 		};
+	},
+	async (req, res): Promise<void> => {
+		console.log(req.body);
+		res.sendStatus(400);
 	}
```
Comment (Scarpy019) [inline source/controllers/ListingController.ts:73]: why async?

File: source/controllers/UserController.ts [hunk 2]
```diff
@@ -96,18 +96,17 @@ login.create = login.handler(
...
+login.delete = async (req, res) => {
 	if (res.locals.user !== null && res.locals.user !== undefined) {
 		res.clearCookie('AuthToken');
-		res.redirect('/section');
+		res.send('Logging out');
 	} else {
 		res.send('Not Logged in');
-		res.redirect('/section');
 	}
 };
```
Comment (Scarpy019) [inline source/controllers/UserController.ts:106]: why async?

**Review by Scarpy019 — APPROVED** [review]
Some functions are needlessly async and I'd like to know the reason but i approve otherwise

File: source/controllers/ListingController.ts (hunk 1 above)
Comment (EPoikans) [reply source/controllers/ListingController.ts:73]: - nevajag
- labs jautajums

File: source/controllers/UserController.ts (hunk 2 above)
Comment (EPoikans) [reply source/controllers/UserController.ts:106]: vairs nebūs

**Review by Scarpy019 — APPROVED** [review]

**Pull request merged**